rust/rsc/src/bin/rsc/types.rs

```diff
@@ -122,6 +122,51 @@ impl ReadJobPayload {
     }
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub struct AllowJobPayload {
+    pub cmd: Vec<u8>,
+    pub env: Vec<u8>,
+    pub cwd: String,
+    pub stdin: String,
+    pub is_atty: bool,
+    pub hidden_info: String,
+    pub visible_files: Vec<VisibleFile>,
+    // Above is the job key and required.
+    // Below is optional and used for determining allowability
+    pub status: i32,
+    pub runtime: f64,
+    pub cputime: f64,
+    pub memory: u64,
+    pub obytes: u64,
+    pub label: String,
+}
+
+impl AllowJobPayload {
+    // TODO: Figure out a way to de-dup this with AddJobPayload somehow
```

Comment:
😬 Looks to be a clean copy paste right now

Reply:
Its a bit trickier than it seems on the surface

The 3 types (Post, Read, Allow) are all subtly different but they do all have the necessariy members to hash _just_ the job key. 

We can't easily have a regular function that does it since they are all different types. Using something like a `trait` would work but IIRC traits only allow you to specify methods so we'd be moving the boilerplate to a bunch of hand copied `fn get_foo_bar()` getter functions.

One thing that would probably work but I didn't consider until now is a generic function but I haven't tried that out yet. It does get rid of the super nice `object.hash()` api we currently have though

Reply:
The current boilerplate is at least much easier to check.